Advanced Search

micahmills opened the pull request:
This gives an option for advanced searching of specific text fields in Contacts and Groups, and lays the groundwork for a new global search endpoint for the future.

corsacca commented on archive-template.php:
```diff
@@ -28,14 +28,67 @@ class="second-bar list-actions-bar">
+                                <?php echo esc_html( in_array( 'all', $fields_to_search ) ? "checked" : '' ); ?>
+
+                                style="margin:0">
+                        <b>Search All Fields</b>
```
Translate

micahmills replied:
I have fixed this

corsacca commented on archive-template.php:
```diff
@@ -50,10 +103,63 @@ class="second-bar show-for-small-only center list-actions-bar"><!--  /* MOBILE V
+                                <?php echo esc_html( in_array( 'all', $fields_to_search ) ? "checked" : '' ); ?>
+
+                                style="margin:0">
+                        <b>Search All Fields</b>
```
translate

micahmills replied:
I have fixed this

corsacca commented:
![image](https://user-images.githubusercontent.com/24901539/108383743-e97de900-71be-11eb-8469-50f32b947cd9.png)
I would expect select one of the options to limit the search to that field. So if comments is selected, it would only search for record with "bob" in the comments.

micahmills commented:
> ![image](https://user-images.githubusercontent.com/24901539/108383743-e97de900-71be-11eb-8469-50f32b947cd9.png)
> I would expect select one of the options to limit the search to that field. So if comments is selected, it would only search for record with "bob" in the comments.

I think I have fixed this issues.

I also added a badge icon and number when an advanced fields search is selected so you can tell if they are toggled without having to open it.

corsacca commented:
Thanks @micahmills !
